api/src/test/java/org/openmrs/util/OpenmrsUtilUnitTest.java

```diff
@@ -137,4 +147,128 @@ public void join_shouldReturnListElementsJoinedBySeparatorIfGivenSeparatorIsNull
 
 		assertThat(OpenmrsUtil.join(moduleList, null), is("module1nullmodule2"));
 	}
+
+	@Test
+	public void isStringInArray_shouldPassIfStringIsInArray() {
+
+		String[] array = new String[]{"element1","element2","element3"};
+
+		assertTrue(OpenmrsUtil.isStringInArray("element1", array));
+	}
+
+	@Test
+	public void isInNormalNumericRange_shouldNotAllowNullValuesAsLowerAndHigherNormal() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiNormal(null);
+
+		assertFalse(OpenmrsUtil.isInNormalNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInNormalNumericRange_shouldPassIfValueWithinRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiNormal(10.34d);
+		concept.setLowNormal(3.67d);
+
+		assertTrue(OpenmrsUtil.isInNormalNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInCriticalNumericRange_shouldNotAllowNullValuesAsLowerAndHigherCritical() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(null);
+
+		assertFalse(OpenmrsUtil.isInCriticalNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInCriticalNumericRange_shouldPassIfValueWithinRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiCritical(10.34d);
+		concept.setLowCritical(3.67d);
+
+		assertTrue(OpenmrsUtil.isInCriticalNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInAbsoluteNumericRange_shouldNotAllowNullValuesAsLowerAndHigherAbsolute() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(null);
+
+		assertFalse(OpenmrsUtil.isInAbsoluteNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isInAbsoluteNumericRange_shouldPassIfValueWithinRange() {
+
+		ConceptNumeric concept = new ConceptNumeric();
+		concept.setHiAbsolute(10.34d);
+		concept.setLowAbsolute(3.67d);
+
+		assertTrue(OpenmrsUtil.isInAbsoluteNumericRange(5.67f, concept));
+	}
+
+	@Test
+	public void isTestMode_shouldReturnTrueIfFunctionalTestModeIsEnabled() {
+
+		System.setProperty("FUNCTIONAL_TEST_MODE", Boolean.toString(true));
+
+		assertTrue(OpenmrsUtil.isTestMode());
+	}
+
+	@Test
+	public void isTestMode_shouldReturnFalseIfFunctionalTestModeIsDisabled() {
+
+		System.setProperty("FUNCTIONAL_TEST_MODE", Boolean.toString(false));
+
+		assertFalse(OpenmrsUtil.isTestMode());
+	}
+
+	@Test
+	public void compareLists_shouldPassIfMapOfDifferencesIsCorrect() {
+
+		LinkedList<Integer> origList = new LinkedList<>(Arrays.asList(1,2,3));
+		LinkedList<Integer> newList = new LinkedList<>(Arrays.asList(2,3,4));
+
+		Collection<Collection<Integer>> m = OpenmrsUtil.compareLists(origList, newList);
+		Vector<Collection<Integer>> v = new Vector<Collection<Integer>>(m);
```

Comment:
can we remove creation of vector and maybe simply do

```java
LinkedList<Integer> origList = new LinkedList<>(Arrays.asList(1,2,3));
LinkedList<Integer> newList = new LinkedList<>(Arrays.asList(2,3,4));

Collection<Collection<Integer>> result = OpenmrsUtil.compareLists(origList, newList);

assertThat(result.size(), is(2));
assertThat(result, contains(1, 4));
```

see http://www.baeldung.com/hamcrest-collections-arrays